### DISCUSSION TITLE
Terminal node size in plot(get.tree()) is smaller than nodesize

### DISCUSSION THREAD

**Discussion by glyden** [kogalur/randomForestSRC#376] [Q&A] [answered]
I thought that the `nodesize` parameter in `rfsrc()` was a lower bound for the terminal node size. That is, I thought a tree could not have a terminal node with fewer observations than `nodesize`.

But after plotting a `get.tree` object for my random forest, I see that the terminal leaf sizes are frequently smaller than my specified `nodesize` of 15. **Can you clarify what `nodesize` means?**

Is it actually the following: A node will not split if the size is < `nodesize`, but it can split if it has at least `nodesize` observations, resulting possibly in terminal nodes that are smaller than `nodesize`? (For example, if a node of size 20 splits into terminal nodes of size 15 and 5.)

**Comment by kogalur** [comment, marked as answer]
A node will be split under the following conditions:

1. The depth of the node must be less than the maximum depth allowed.
2. The node must have some impurity to proceed, otherwise we prematurely terminate the node.
3. The number of replicates must be greater than or equal to 2 x nodesize.

If all these conditions are satisfied, we split the node.  The result is that some nodes will be greater than nodesize and some will be larger, depending on where the split is, but "on average", the terminal nodes will be of size nodesize.

**Reply by glyden** [reply]
Great answer, thank you! And by replicates, do you mean observations / individuals / rows of the dataset?

**Reply by kogalur** [reply]
We mean the number of individuals in the bootstrap sample in the node.
